### PR TITLE
Migrate to .net 5 / 6 / Optimizely + localization

### PR DESCRIPTION
**Pull request by ellinge**
image

A basic migration to .net 5 / 6 with localization added. Seems to be working but I don't have any experience with the product from previous versions so not sure. 

Perhaps one should change it to be non commerce specific to be able to specify roots with an attribute and be aware of AllowedTypes.

### CODE REVIEW

**Comment by ellinge** [comment]
Removed the dependency on Commerce and added the possibility to specify allowed types and roots for selecting other stuff. Fixing #9 
```
    [Roots(RootSelection.StartPage | RootSelection.CommerceRootPage)]
    [Types(typeof(PageData), typeof(ProductContent))]
```
image

File: .gitignore [hunk 1]
```diff
@@ -1,39 +1,44 @@
-## Ignore Visual Studio temporary files, build results, and
```
Comment (marisks) [inline .gitignore:1]: This gitignore doesn't look standard one. Should we update to a common gitignore and add project-specific things at the end?

**Comment by marisks** [comment]
We definitely need a new repository.

File: src/Geta.Optimizely.HotspotsEditor/Geta.Optimizely.HotspotsEditor.csproj [hunk 2]
```diff
@@ -0,0 +1,51 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net5.0;net6.0</TargetFrameworks>
```
Comment (marisks) [inline src/Geta.Optimizely.HotspotsEditor/Geta.Optimizely.HotspotsEditor.csproj:4]: Just use net6.0 and the first Optimizely version that supports it.

P.S. Old Optimizely versions have some security holes, so better force upgrade for everyone. Also, net5 is dead - no need to support it.

File: src/pack.proj [hunk 3]
```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (marisks) [inline src/pack.proj:1]: Let's use common conventions - packaging the same way as in other of our projects.

**Review by marisks — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (ellinge) [reply .gitignore:1]: I think I based it upon one of your other projects:
https://github.com/Geta/geta-optimizely-genericlinks/blob/master/.gitignore

File: src/pack.proj (hunk 3 above)
Comment (ellinge) [reply src/pack.proj:1]: I based it upon one of your other projects:
https://github.com/Geta/geta-optimizely-genericlinks/blob/master/src/pack.proj

**Comment by ellinge** [comment]
> We definitely need a new repository.

A rename was suggested in #10

File: .gitignore (hunk 1 above)
Comment (ellinge) [reply .gitignore:1]: Changed it to be based upoon https://github.com/Geta/geta-notfoundhandler/blob/master/.gitignore instead

File: src/pack.proj (hunk 3 above)
Comment (ellinge) [reply src/pack.proj:1]: Tried to check https://github.com/Geta/geta-notfoundhandler for inspiration, the reason for the pack.proj seems to be handle module zip (ClientResources) which notfoundhandler lacks.

Comment (marisks) [reply src/pack.proj:1]: Sitemaps is a better example: https://github.com/Geta/geta-optimizely-sitemaps

**Comment by marisks** [comment]
> > We definitely need a new repository.
> 
> A rename was suggested in #10

Rename is not an option. We need a separate repository. But after migration is done in this PR, we will create a repo and will move the new code there.

File: src/pack.proj (hunk 3 above)
Comment (ellinge) [reply src/pack.proj:1]: https://github.com/Geta/geta-optimizely-sitemaps/blob/master/src/Geta.Optimizely.Sitemaps/msbuild/Main.proj so you want it here instead?

Comment (marisks) [reply src/pack.proj:1]: Yes, with this file name and with cleaner csproj.

**Comment by ellinge** [comment]
History gone bye bye 🙄

**Comment by marisks** [comment]
Yes, because that is a new package, so new history.
The history for the old package will be here anyways.

**Comment by ellinge** [comment]
net6 merged into master of new github repo with history kept:
```
git clone https://github.com/Geta/Geta.EPi.HotspotsEditor.git
cd .\Geta.EPi.HotspotsEditor\
git checkout net6
git remote add neworigin https://github.com/ellinge/newtestrep.git
git push -u neworigin net6:master
```

**Pull request merged**